File: apps/api/app/schemas/risk_analysis.py
import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel


class RiskFindingResponse(BaseModel):
    id: uuid.UUID
    category: str
    severity: str
    title: str
    reason: str
    confidence: float
    citations: list[dict[str, Any]]

    model_config = {"from_attributes": True}


class RiskAnalysisResponse(BaseModel):
    id: uuid.UUID
    document_id: uuid.UUID
    status: str
    risk_score: int | None
    error_message: str | None
    created_at: datetime
    findings: list[RiskFindingResponse]

    model_config = {"from_attributes": True}

File: apps/web/types/risk-analysis.ts
import type { Citation } from "@/types/chat";

export type RiskSeverity = "high" | "medium" | "low";

export interface RiskFinding {
  id: string;
  category: string;
  severity: RiskSeverity;
  title: string;
  reason: string;
  confidence: number;
  citations: Citation[];
}

export interface RiskAnalysis {
  id: string;
  document_id: string;
  status: "running" | "completed" | "failed";
  risk_score: number | null;
  error_message: string | null;
  created_at: string;
  findings: RiskFinding[];
}
